dds_web/web/user.py

```diff
@@ -339,18 +346,29 @@ def reset_password(token):
         return flask.redirect(flask.url_for("auth_blueprint.index"))
 
     # Verify that the token is valid and contains enough info
-    user = models.User.verify_reset_token(token=token)
-    if not user:
+    try:
+        user = dds_web.security.auth.verify_password_reset_token(token=token)
+    except ddserr.AuthenticationError:
         flask.flash("That is an invalid or expired token", "warning")
-        return flask.redirect(flask.url_for("auth_blueprint.request_reset_password"))
+        return flask.redirect(flask.url_for("auth_blueprint.index"))
 
     # Get form for reseting password
     form = forms.ResetPasswordForm()
 
     # Validate form
     if form.validate_on_submit():
+        # Delete project user keys for user
+        for project_user_key in user.project_user_keys:
+            db.session.delete(project_user_key)
+        db.session.commit()
+
+        # Update user password
         user.password = form.password.data
         db.session.commit()
+
+        dds_web.security.project_user_keys.generate_user_key_pair(user)
```

Comment:
Please don't generate_user_key_pair here, instead
```
user.kd_salt = None
user.nonce = None
user.public_key = None
user.private_key = None

user.password = form.password.data
db.session.commit()
```

Reply:
So when would it be generated?

Reply:
It's in my key derivation PR. Keys are generated if the lines that I wrote above come in that order. Key generation is triggered from `user.password = form.password.data`

Reply:
I can change that then. I will need to wait for your PR to have the tests pass though.

Reply:
Ok, fixed now!